src/sidebar/services/test/streamer-test.js

```diff
@@ -264,21 +267,19 @@ describe('StreamerService', () => {
       console.error.restore();
     });
 
-    it('should reconnect when user changes', () => {
-      let oldWebSocket;
+    it('should reconnect when user changes', async () => {
       createDefaultStreamer();
 
-      return activeStreamer
-        .connect()
-        .then(() => {
-          oldWebSocket = fakeWebSocket;
-          fakeStore.profile.returns({ userid: 'somebody' });
-          return fakeStore.setState({});
-        })
-        .then(() => {
-          assert.ok(oldWebSocket.didClose);
-          assert.ok(!fakeWebSocket.didClose);
-        });
+      await activeStreamer.connect();
+
+      const oldWebSocket = fakeWebSocket;
+      fakeStore.profile.returns({ userid: 'somebody' });
+      fakeStore.setState({});
```

Comment:
Tangential, but I wonder about the strategy of setting an empty state for the entire `store` in tests; maybe it's fine but it seems to break out of the unit we are testing...I know this wasn't part of these changes.

Reply:
I believe this code was written in the early days of Redux usage in the client, when we hadn't settled on the modern API that the store has and associated patterns for writing tests.